### PR TITLE
Update navigationbar to be a react component

### PR DESCRIPTION
**Pull request by lifeiscontent**
@justin808 what's next? :)

[Review on Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/231)

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -19,32 +19,7 @@
...
-        <li><%= link_to "Simple React", simple_path %></li>
-        <li><%= link_to "Classic Rails", comments_path %></li>
-        <li><%= link_to "Source on Github", "https://github.com/shakacode/react-webpack-rails-tutorial" %></li>
-        <li><%= link_to "Tutorial Article", "http://www.railsonmaui.com/blog/2014/10/03/integrating-webpack-and-the-es6-transpiler-into-an-existing-rails-project/" %></li>
-        <li><%= link_to "Forum Discussion", "http://forum.shakacode.com/t/fast-rich-client-rails-development-with-webpack-and-the-es6-transpiler/82/22" %></li>
-      </ul>
-    </div><!-- /.navbar-collapse -->
-  </div><!-- /.container-fluid -->
-</nav>
+  <%= react_component 'Navigationbar', props: {} %>
```
Comment (lifeiscontent) [inline app/views/layouts/application.html.erb]: @justin808 I couldn't render this component without passing the blank props... this seems odd.

Comment (justin808) [inline app/views/layouts/application.html.erb]: That might be the case. I thought we default to blank in the view helper.

Comment (justin808) [inline app/views/layouts/application.html.erb]: We should file an issue and I should fix it if that is the case.

**Comment by justin808** [comment]
All looks good so far on code review, but CI is failing, so something is not quite right. Let's fix the CI issue, and the next step is to use the ReactOnRails.registerStore to create a store that is shared by both the top navbar and bottom content areas.

See https://github.com/shakacode/react_on_rails/blob/master/spec%2Fdummy%2Fapp%2Fviews%2Fpages%2Fserver_side_hello_world_shared_store.html.erb for an example of how this is done.

**Comment by aaronvb** [comment]
@lifeiscontent It looks like eslint errors are causing the CI to fail.

```
[4m/home/travis/build/shakacode/react-webpack-rails-tutorial/client/app/bundles/comments/components/Navigationbar/Navigationbar.jsx[24m
   [90m1:17[39m  [31merror[39m  "PropTypes" is defined but never used           [90mno-unused-vars[39m
  [90m14:1[39m   [31merror[39m  Line 14 exceeds the maximum line length of 100  [90mmax-len[39m

[4m/home/travis/build/shakacode/react-webpack-rails-tutorial/client/app/bundles/comments/startup/clientRegistration.jsx[24m
  [90m16:18[39m  [31merror[39m  Missing trailing comma  [90mcomma-dangle[39m

[31m[1mâœ– 3 problems (3 errors, 0 warnings)
```

**Comment by aaronvb** [comment]
@lifeiscontent `rake lint:eslint ` in the root react-webpack-rails-tutorial should show you any errors you have.

**Comment by lifeiscontent** [comment]
@aaronvb the next lint failure is almost impossible to fix without removing the URLs to the navigationbar...

**Comment by aaronvb** [comment]
@lifeiscontent There are examples I've seen in the React docs where they use strings in javascript and combine them.

```jsx
<li>
  <a href={'https://github.com/shakacode/' +
    'react-webpack-rails-tutorial'}>
    Source on Github
  </a>
</li>
```

AFAIK, there's no formal way of doing this(please correct me if I'm wrong), and not much info out there.

**Comment by alexfedoseev** [comment]
@lifeiscontent @aaronvb Long strings in jsx is a pain. One of the possible workarounds: assign string to variable and [disable eslint rule](http://eslint.org/docs/user-guide/configuring#configuring-rules) for this line.

**Comment by aaronvb** [comment]
@alexfedoseev I found that we can have the max length ignore URLs in eslint: 

```
max-len: [2, 80, 4, {"ignoreUrls": true}]
```

http://eslint.org/docs/rules/max-len#options

**Comment by justin808** [comment]
@aaronvb can you make this a PR to 

https://github.com/shakacode/style-guide-javascript/tree/master/packages/eslint-config-shakacode

@alexfedoseev We might need something for eslint as well

@aaronvb Let me know if you get stuck. We want to get this finished by the weekend.

**Comment by aaronvb** [comment]
@justin808 Sure thing! We'll have to get eslint-config-shakacode into this tutorial. Shall I do a PR for that as well?

**Comment by aaronvb** [comment]
@justin808 My bad that line is for eslint, which `eslint-config-airbnb` sets to ignore URLs by default. The error  happening with the CI is with `jscs` and unfortunately doesn't have that option.

**Comment by lifeiscontent** [comment]
@justin808 what's next?

**Comment by justin808** [comment]
@lifeiscontent @seoyoochan Here's what we need to do:

1. Convert first 2 tabs to use a shared store
2. Have the navbar component also use the shared store, and have the navbar display the count of comments in the store.

When a new comment is added, the navbar should automatically update.

When you go to a different tab (without the store), we can either:

1. use a different navbar (not DRY).
2. Have some logic to check if the store is defined on the page. A simple check for the dom class or id can do that. Note, with turbolinks, the state of the store should stay around as you go from tab to tab.

Study this example:

* https://github.com/shakacode/react_on_rails/blob/master/spec%2Fdummy%2Fapp%2Fviews%2Fpages%2Fserver_side_hello_world_shared_store.html.erb#L31
* https://github.com/shakacode/react_on_rails/search?utf8=%E2%9C%93&q=SharedReduxStore

**Comment by justin808** [comment]
See #239. Contains this code and more.

**Pull request closed without merging**